### PR TITLE
Check and close empty sessions on creation

### LINKED ISSUE #14: Newly created sessions aren't deleted if nobody actually joins them.
If somebody makes a request to `/createSession` but then doesn't join the newly created session, said session is never deleted. This is because the check for emptiness currently only takes place when a user leaves a session.

There should either be a regular check for all (or only newly created) sessions that removes them if empty, or session creation should be moved to when it is actually joined. It might make most sense to just add a timeout to every new session that deletes it again if nobody joins within something like a minute. That way, the interval is always the same for each session.

### PR DESCRIPTION
**Pull request by s-gbz**
Closes #14 

# Previous behaviour:
session-not-closing

# New behaviour:
sessions-closing-if-empty

### CODE REVIEW

File: src/server/Session.ts [hunk 1]
```diff
@@ -25,3 +27,18 @@
     },
   }
 }
+
+export function closeSessionIfEmpty(session: Session, sessionID: string, sessions: SessionsObject) {
+  if (session.webSockets.size === 0) {
+    console.log(`Session ${sessionID} is now empty. Waiting ${EMPTY_SESSION_TIMEOUT} seconds for potential re-joins before deleting it`)
```
Comment (LBBO) [inline src/server/Session.ts:33]: This console.log statement is no longer accurate, as the session might have just been created (so it isn't suddenly empty and we aren't waiting for re-joins but rather initial joins). I would probably add an optional parameter to toggle between two messages here.

File: src/server/SessionManager/SessionManager.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export const sessionManager = (sessions: SessionsObject): WebsocketRequestHandle
   const session = sessions[sessionID]
 
   const intervalID = setInterval(() => {
-    // Only send ping if socked hasn't closed in the mean time
+    // Only send ping if socket hasn't closed in the mean time
```
Comment (LBBO) [inline src/server/SessionManager/SessionManager.ts:11]: Thanks for catching that one :D

File: src/server/Session.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import { v4 as createUuid } from 'uuid'
 import * as ws from 'ws'
 
+const EMPTY_SESSION_TIMEOUT = 90
```
Comment (LBBO) [inline src/server/Session.ts:4]: I think there should be either a comment or a different variable name here to clarify the timeout is in seconds, not milliseconds. Just to be extra clear

**Review by LBBO — CHANGES_REQUESTED** [review]
Thank you for your PR! All in all, it's awesome and exactly what I was hoping for. I have two minor issues though that I'd like fixed before merging.

**Comment by s-gbz** [comment]
I thought about it and suggest to handle the session closing in respective cases.
Benefits:
- Testability
- Readability
- Expandability

Don't get me started how the *function might have looked if it was left joined. I hope this works, feel free to add further suggestions!

**Review by LBBO — APPROVED** [review]

**Comment by LBBO** [comment]
Perfect. Thank you for your work!

**Pull request merged**